lib/gpio.ex

```diff
@@ -292,6 +292,19 @@ defmodule Circuits.GPIO do
   def enumerate(nil), do: enumerate(default_backend())
   def enumerate({backend, _options}), do: backend.enumerate()
 
+  @doc """
+  Checks if a spec is valid. Useful for storing specs in `config.exs` or similar
+  and evaluating at runtime.
+  """
+  @spec spec_valid?(any()) :: boolean()
+  def spec_valid?({chip, pin})
+      when (is_binary(chip) or is_integer(chip)) and (is_binary(pin) or is_integer(pin)),
+      do: true
+
+  def spec_valid?(label) when is_binary(label), do: true
+  def spec_valid?(pin) when is_integer(pin), do: true
+  def spec_valid?(_), do: false
```

Comment:
The code looks ugly, but it seems like this could be a guard and that would let it be used more places.

```suggestion
  @spec is_gpio_spec(any()) :: boolean()
  defguard is_gpio_spec(x) when (is_tuple(x) and is_binary(elem(x, 0)) and (is_binary(elem(x, 1)) or is_integer(elem(x, 1))) or is_binary(x) or is_integer(x)
```